pokemonterminal/platform/named_event/adapters/posix.py

```diff
@@ -0,0 +1,47 @@
+import posix_ipc
+
+from . import NamedEvent
+
+class PosixNamedEvent(NamedEvent):
+    """
+    A wrapper for named events using a named semaphore
+    """
+
+    def __init__(self, name: str):
+        semaphore_name = '/' + name
+        try:
+            self.__semaphore = posix_ipc.Semaphore(semaphore_name, flags=posix_ipc.O_CREX)
```

Comment:
Yeah I kinda wish Linux had named events which would made this simplier (it only has anonymous events). The named event works fine on Windows according to my testing, so we could just drop the named semaphore idea completely and use a fifo or a simple file with a deletion watcher (I'd personally drop them under `/var/lock` or `/run/lock` instead of the user home like it was done before)